src/ImageCore.jl

```diff
@@ -33,13 +33,10 @@ const RRArray{To,From,N,M,P} = Base.ReshapedArray{To,N,Base.ReinterpretArray{To,
 const RGArray = Union{Base.ReinterpretArray{<:AbstractGray,M,<:Number,P}, Base.ReinterpretArray{<:Number,M,<:AbstractGray,P}} where {M,P}
 
 # delibrately not export these constants to enable extensibility for downstream packages
-const NumberLike{T<:Number} = Union{T,AbstractGray{T}}
-const RealLike{T<:Real} = NumberLike{T}
-const FloatLike{T<:AbstractFloat} = RealLike{T}
-const FractionalLike{T<:Union{FixedPoint, AbstractFloat}} = RealLike{T}
-const GrayLike{T<:Union{Bool, FixedPoint, AbstractFloat}} = RealLike{T}
-const GenericGrayImage{N, T<:GrayLike} = AbstractArray{<:GrayLike{T}, N}
-const Gray2dImage{T<:GrayLike} = GenericGrayImage{2, T}
+const NumberLike = Union{<:Number,<:AbstractGray}
```

Comment:
I probably messed this up. Should be `const NumberLike = Union{Number,AbstractGray}` (without the `<:`). Sorry!

Reply:
Hmmm, looks like they are equivalent

```julia
julia> Union{<:Number, <:AbstractGray} == Union{Number, AbstractGray}
true
```

Reply:
```julia
julia> Union{<:Number,<:AbstractGray} === Union{Number,AbstractGray}
false
```
(note the three `=`)

Reply:
And yet
```julia
julia> Union{AbstractGray,Number} === Union{Number,AbstractGray}
true
```